### PR TITLE
Issue#458 Support for IN1/IN2 in VXU_V04

### PR DESCRIPTION
**Pull request by cragun47**
Signed-off-by: Brian Cragun <cragun@us.ibm.com>

Support and tests for IN1/IN2 segment support in VXU_V04.

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java [hunk 1]
```diff
@@ -74,24 +74,34 @@ void test_VXU_with_patient_group_that_has_minimum_segments() throws IOException
     }
 
     @Test
-    void test_VXU_with_patient_group_that_has_all_segments() throws IOException {
+    void test_VXU_with_patient_group_that_has_all_segments_and_insurance() throws IOException {
```
Comment (lisadier) [inline src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java:77]: With the addition of insurance here, let's rename this test_VXU_with_full_patient_group_and_insurance.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java [hunk 2]
```diff
@@ -697,7 +697,7 @@ void testInsuranceCoverageOfSelfAndTenant() throws IOException {
     // This test is to see the problem does not return.  It fails without PR #405, and is successful with it. 
     void testFailingOrganizationTelecom() throws IOException {
 
-        String hl7message = "MSH|^~\\&|||||20211214105741||DFT^P03|1760487765|P|2.6|||||||||\n"
+        String hl7message = "MSH|^~\\&|||||20211214105741||DFT^P03^DFT_P03|1760487765|P|2.6|||||||||\n"
```
Comment (lisadier) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:700]: Was there a reason that you added the structure?  I think it is good to test it both ways to ensure that the converter works correctly without the structure provided; this checks that we default the structure correctly.

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java [hunk 3]
```diff
@@ -172,11 +182,15 @@ void test_VXU_with_minimum_patient_group_plus_order_group_with_OBX_but_no_observ
 }
 
     @Test
-    void test_VXU_with_minimum_patient_group_plus_order_group_with_OBX_with_observations() throws IOException {
+    void test_VXU_with_minimum_patient_group_plus_coverage_plus_order_group_with_OBX_with_observations() throws IOException {
```
Comment (lisadier) [inline src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java:185]: This looks great

**Review by lisadier — CHANGES_REQUESTED** [review]
Brian, the code looks great, I just had a couple questions/comments above.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java (hunk 2 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:700]: It just seemed consistent with other tests.  I will remove it so we test both ways.

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java (hunk 1 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java:77]: Done

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java (hunk 3 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/message/Hl7VXUMessageTest.java:185]: Thx.

File: src/main/resources/hl7/message/VXU_V04.yml [hunk 4]
```diff
@@ -46,3 +48,13 @@ resources:
...
               - PATIENT.PV1
+
+    - resourceName: Coverage
+      segment: .IN1
+      group: INSURANCE
+      resourcePath: resource/Coverage
+      repeats: true
+      additionalSegments:
+        - MSH
+        - PID
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/message/VXU_V04.yml]: Checked.  You are correct, it is not needed.  I have removed from other messages with Coverage.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java [hunk 5]
```diff
@@ -933,7 +933,7 @@ void testInsuranceCoverageFromIN2Self() throws IOException {
     // Tests non-related subscriber, employer.
     void testInsuranceCoverageOfWorkersComp() throws IOException {
 
-        String hl7message = "MSH|^~\\&|TEST|TEST|||20220101120000||DFT^P03|1234|P|2.6\n"
+        String hl7message = "MSH|^~\\&|TEST|TEST|||20220101120000||DFT^P03^DFT_P03|1234|P|2.6\n"
```
Comment (lisadier) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:936]: Let's take the structure off here too.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java [hunk 6]
```diff
@@ -1015,7 +1015,7 @@ void testInsuranceCoverageOfWorkersComp() throws IOException {
     // As part of the test, there is a valid code in IN2.72 that would create a related person, but it should be ignored.
     void testInsuranceCoverageUnknownRelationship() throws IOException {
 
-        String hl7message = "MSH|^~\\&|TEST|TEST|||20220101120000||DFT^P03|1234|P|2.6\n"
+        String hl7message = "MSH|^~\\&|TEST|TEST|||20220101120000||DFT^P03^DFT_P03|1234|P|2.6\n"
```
Comment (lisadier) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:1018]: Let's take the structure off here too.

**Review by lisadier — APPROVED** [review]

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java (hunk 5 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:936]: Done.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java (hunk 6 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:1018]: Done.

**Pull request merged**